File: backend/api/app/models/spark_reward.rb
# frozen_string_literal: true

class SparkReward < ApplicationRecord
  belongs_to :user
  belongs_to :spark_session

  enum :reward_type, { premium_week: 0, match_credit: 1, boost: 2 }
  enum :status,      { pending: 0, redeemed: 1, expired: 2 }

  scope :active, ->(now = Time.current) { where(status: :pending).where("valid_until > ?", now) }
end

File: apps/ios/Synca/Models/SparkReward.swift
import Foundation

enum SparkRewardType: String, Codable {
    case premiumWeek = "premium_week"
    case matchCredit = "match_credit"
    case boost
}

enum SparkRewardStatus: String, Codable {
    case pending, redeemed, expired
}

/// `GET /spark_rewards` — full reward record (`id`/`valid_until` included, unlike
/// the summary embedded in `SparkResult`).
struct SparkReward: Codable, Equatable, Identifiable {
    let id: Int
    let rewardType: SparkRewardType
    let status: SparkRewardStatus
    let validUntil: Date?
}

struct SparkRewardListResponse: Codable {
    let rewards: [SparkReward]
}
